upup/pkg/fi/cloudup/template_functions.go

```diff
@@ -156,29 +156,28 @@ func (tf *TemplateFunctions) DnsControllerArgv() ([]string, error) {
 	}
 
 	if dns.IsGossipHostname(tf.cluster.Spec.MasterInternalName) {
-		argv = append(argv, "--dns=gossip")
 		argv = append(argv, "--gossip-seed=127.0.0.1:3999")
-	} else {
-		switch kops.CloudProviderID(tf.cluster.Spec.CloudProvider) {
-		case kops.CloudProviderAWS:
-			if strings.HasPrefix(os.Getenv("AWS_REGION"), "cn-") {
-				argv = append(argv, "--dns=gossip")
-			} else {
-				argv = append(argv, "--dns=aws-route53")
-			}
-		case kops.CloudProviderGCE:
-			argv = append(argv, "--dns=google-clouddns")
-		case kops.CloudProviderDO:
-			// this is not supported yet, here so we can successfully create clusters
-			// this will be supported for digitalocean in the future
-			argv = append(argv, "--dns=digitalocean")
-		case kops.CloudProviderVSphere:
-			argv = append(argv, "--dns=coredns")
-			argv = append(argv, "--dns-server="+*tf.cluster.Spec.CloudConfig.VSphereCoreDNSServer)
-
-		default:
-			return nil, fmt.Errorf("unhandled cloudprovider %q", tf.cluster.Spec.CloudProvider)
+	}
+
+	switch kops.CloudProviderID(tf.cluster.Spec.CloudProvider) {
+	case kops.CloudProviderAWS:
+		if strings.HasPrefix(os.Getenv("AWS_REGION"), "cn-") {
+			argv = append(argv, "--dns=gossip")
```

Comment:
Off topic from this PR, but @qqshfox does CN regions have route53 now?

Reply:
Unfortunately still no route53 in CN regions.